src/h5web/explorer/utils.ts

```diff
@@ -9,83 +9,110 @@ import {
   MyHDF5Datatype,
   HDF5HardLink,
   MyHDF5Metadata,
+  MyHDF5EntityKind,
+  HDF5Datatype,
+  HDF5Dataset,
 } from '../providers/models';
-import {
-  makeMyDataset,
-  makeMyDatatype,
-  makeMyGroup,
-  makeMyLink,
-} from '../providers/my-utils';
-import { isGroup, isReachable } from '../providers/utils';
+import { isGroup, isReachable } from '../guards';
 import { getChildEntity } from '../visualizations/nexus/utils';
+import { nanoid } from 'nanoid';
 
-function buildDataset(
-  metadata: HDF5Metadata,
-  link: HDF5HardLink
-): MyHDF5Dataset {
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I've managed to get rid of these ESLint comments by changing the signature of `buildGroup` to accept a metadata object of type `Required<HDF5Metadata>`. I prepare `HDF5Metadata` to fit this type in `buildTree` below.